scrowl-io/scrowl
# Sc 996 updating active slide

### Short summary

Originally I didn't have IDs to work with and was just using the names of modules, lessons and slides to match against the activeSlide. I've since updated the mock-data to use IDs, and then use more of a findByID approach in the right-pane-content-form.tsx, also cutting down on the amount of looping I needed to do with the old method. This works for editing existing slides (and works after renaming slides, or having multiple slides with the same name), but will not work for newly-created slides--for this to work, we'll need to add IDs to slides as they're saved.

---

### Test steps

1. Open a slide from the left pane -> course outline
2. edit content of fields in any combination
3. click Save
4. open another slide from the course outline: form should be re-populated with initial values
5. edit content of fields
6. click Save
7. save at the application level (CMD+S or application menu)
8. refresh the app, go back into the project and open each of the edited slides; switching between the slides should demonstrate that the edited values were saved and are now rendered out in their respective slides
